samples/MemoryCacheFileWatchSample/Program.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.FileProviders;
+
+namespace MemoryCacheFileWatchSample
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var cache = new MemoryCache(new MemoryCacheOptions());
+            var greeting = "";
+            var cacheKey = "cache_key";
+            var fileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), "WatchedFiles"));
+
+            while (true)
+            {
+                if (!cache.TryGetValue(cacheKey, out greeting))
+                {
+                    greeting = File.ReadAllText((Path.Combine(Directory.GetCurrentDirectory(), "WatchedFiles", "example.txt")));
+                    cache.Set(cacheKey, greeting, new MemoryCacheEntryOptions()
+                         //Telling the cache to depend on the IChangeToken from watching examples.txt
+                         .AddExpirationToken(fileProvider.Watch("example.txt"))
+                         .RegisterPostEvictionCallback(
+                         (echoKey, value, reason, substate) =>
+                         {
+                             Console.WriteLine($"{echoKey} : {value} was evicted due to {reason}");
+                         }));
+                    Console.WriteLine($"{cacheKey} updated from source.");
+                }
+                else
+                {
+                    Console.WriteLine($"{cacheKey} retrieved from cache.");
+                }
+
+                Console.WriteLine(greeting);
+                Console.ReadKey();
```

Comment:
Two things here:
`Console.ReadKey(bool)` can optionally hide the input key.
Might also want to have a graceful way of exiting the infinite loop. If Ctrl-C doesn't work, you might want to consider recognizing a specific key (ESC maybe?) to exit. Print a message like "Press {key} to exit." if you do.